src/lib.rs

```diff
@@ -181,6 +181,88 @@ pub enum Attr {
     BackgroundColor(color::Color)
 }
 
+/// An error arising from interacting with the terminal.
+#[derive(Debug)]
+pub enum Error {
+    /// Indicates an error from any underlying IO
+    Io(io::Error),
+    /// Indicates an error during terminfo parsing
+    TerminfoParsing(terminfo::Error),
+    /// Indicates an error expanding a parameterized string from the terminfo database
+    ParameterizedExpansion(terminfo::parm::Error),
+    /// Indicates that the terminal does not support the requested attribute
+    AttributeNotSupported,
+    /// Indicates that the `TERM` environment variable was unset, and thus we were unable to detect
+    /// which terminal we should be using.
+    TermUnset,
+    /// Indicates that we were unable to find a terminfo entry for the requested terminal.
+    TerminfoEntryNotFound,
+    /// Indicates that the cursor could not be moved to the requested position.
+    CursorDestinationInvalid,
+    /// Indicates that the terminal does not support the requested color.
+    ///
+    /// This is like `AttributeNotSupported`, but more specific.
+    ColorNotSupported,
```

Comment:
`ColorOutOfRange` seems a bit weird to me as a name, since I usually think of ranges as approximating some continuous monotonic spectrum, which these colors don't. But, I can't think of a better name and agree that this is a bad name.